OpenRA.Mods.Common/Traits/BotModules/McvManagerBotModule.cs

```diff
@@ -132,14 +132,31 @@ void IBotTick.BotTick(IBot bot)
 
 		bool ShouldBuildMCV()
 		{
+			// Only build MCV if we have a factory to build it with.
+			var enoughFactories = AIUtils.CountBuildingByCommonName(Info.McvFactoryTypes, player) > 0;
+			if (!enoughFactories)
+				return false;
+
 			// Only build MCV if we don't already have one in the field.
-			var allowedToBuildMCV = AIUtils.CountActorByCommonName(Info.McvTypes, player) == 0;
-			if (!allowedToBuildMCV)
+			var mcvCount = AIUtils.CountActorByCommonName(Info.McvTypes, player);
+			if (mcvCount > 0)
+				return false;
+
+			// Only build MCV if the conyardCount + mcvCount are lower than the minimum conyard count.
```

Comment:
This comment now wants the reference to `mcvCount` removed as well, I suppose.

Thinking about it, you should remove the first three comments (line 135, 140 and this one) as Imho your variables are well named so it is easy to understand what is going on already (i.e. the comments serve no purpose).